static.js

```diff
@@ -0,0 +1,86 @@
+'use strict'
+
+const assert = require('assert')
+const path = require('path')
+const fs = require('fs')
+const yaml = require('js-yaml')
+
+module.exports = function (fastify, opts, next) {
+  assert(opts.specification, 'specification is missing in the module options')
+  assert(opts.specification.type, 'specification.type is missing')
+  assert(
+    typeof opts.specification.type === 'string',
+    'specification.type is not a string'
+  )
+  opts.specification.type = opts.specification.type.toLowerCase()
+  assert(
+    ['file'].indexOf(opts.specification.type) !== -1,
+    "specification.type is incorrect, should be one from ['file']"
+  )
+
+  let swaggerObject = {}
+
+  if (opts.specification.type === 'file') {
+    assert(
+      opts.specification.path,
+      'specification.path is missing, should be path to the file'
+    )
+    assert(
+      typeof opts.specification.path === 'string',
+      'specification.path is not a string'
+    )
+    assert(
+      fs.existsSync(`${process.cwd()}/${opts.specification.path}`),
+      `${opts.specification.path} does not exist`
+    )
+    const extName = path.extname(opts.specification.path).toLowerCase()
+    assert(
+      ['.yaml', '.json'].indexOf(extName) !== -1,
+      "specification.path extension name is not supported, should be one from ['.yaml', '.json']"
+    )
+
+    // read
+    const source = fs.readFileSync(
+      `${process.cwd()}/${opts.specification.path}`,
```

Comment:
This is not cross platform, you should use `path.resolve` instead.
Same above :)

Reply:
updated, thanks